core/src/main/java/org/apache/iceberg/V1Metadata.java

```diff
@@ -213,7 +213,8 @@ static Schema wrapFileSchema(Types.StructType fileSchema) {
         DataFile.LOWER_BOUNDS,
         DataFile.UPPER_BOUNDS,
         DataFile.KEY_METADATA,
-        DataFile.SPLIT_OFFSETS
+        DataFile.SPLIT_OFFSETS,
+        ManifestFile.SPEC_ID.asOptional()
```

Comment:
This is the v1 write schema. Since we don't need to write spec IDs into manifests, I don't think that we need to add this here or to update to add it below.

Reply:
The issue with this is that we have tests which compare schemas, if the field is not listed here it still gets included from Datafile which makes the comparisons
"partition_spec_id" vs "partition_spec_id_r502" which fails. So I mostly added it here so the test would pass

Reply:
I'll remove this when we switch to the isolated tests for this action, rather than me testing on just all possible tests and making sure nothing breaks